### PR TITLE
[IMP] hw_drivers: landscape, recto-verso printing options

### PR DESCRIPTION
**Pull request by loouis-t**
added more options to print via cups:
- duplex mode (recto/verso),
- landscape mode.

Task: 3598690

### CODE REVIEW

**Comment by loouis-t** [comment]
@odoo/rd-security Could you have a look at this ?

## Context 

We want to enable more CUPS printing options for future developments.

### Solution proposed 

We add two more parameters to the method used for printing with default values.  
Depending on them, we build a dictionary with CUPS corresponding parameters.

## Security issues

- `process = subprocess.Popen(["lp", "-d", self.device_identifier, *options], stdin=subprocess.PIPE)`: We need to send the dictionary through a subprocess call. 
  - **Safe because:** All parameters are hardcoded.

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py [hunk 1]
```diff
@@ -172,11 +172,22 @@ def send_status(self):
...
+        :param data: The data to print
+        :param landscape: Print in landscape mode (Default: False)
+        :param duplex: Print in duplex mode (recto-verso) (Default: True)
+        """
+        options = []
+        if landscape:
+            options.extend(['-o', 'orientation-requested=4'])
+        if not duplex:
+            options.extend(['-o', 'sides=one-sided'])
+        process = subprocess.Popen(["lp", "-d", self.device_identifier, *options], stdin=subprocess.PIPE)
```
Comment (lse-odoo) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py:187]:
```suggestion
        cmd = ["lp", "-d", self.device_identifier, *options]
        _logger.debug(..., cmd)
        process = subprocess.Popen(cmd, stdin=subprocess.PIPE)
```

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py [hunk 2]
```diff
@@ -172,11 +172,22 @@ def send_status(self):
         }
         event_manager.device_changed(self)
 
-    def print_raw(self, data):
-        process = subprocess.Popen(["lp", "-d", self.device_identifier], stdin=subprocess.PIPE)
+    def print_raw(self, data, landscape=False, duplex=True):
```
Comment (lse-odoo) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py:175]: is there another PR altering the functional calls ?

**Review by lse-odoo — COMMENTED** [review]
does it make sense to do it this way ? The page orientation can be set in Odoo's report. CUPS should automatically print a received landscape PDF in my opinion.

**Comment by mart-e** [comment]
@robodoo override=ci/security
Reason: construct lp command

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py (hunk 2 above)
Comment (loouis-t) [reply addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py:175]: not for the moment

**Review by lse-odoo — APPROVED** [review]

File: addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py [hunk 3]
```diff
@@ -172,11 +172,25 @@ def send_status(self):
...
+        Print raw data to the printer
+        :param data: The data to print
+        :param landscape: Print in landscape mode (Default: False)
+        :param duplex: Print in duplex mode (recto-verso) (Default: True)
+        """
+        options = []
+        if landscape:
+            options.extend(['-o', 'orientation-requested=4'])
+        if not duplex:
+            options.extend(['-o', 'sides=one-sided'])
```
Comment (Ysoroko) [inline addons/hw_drivers/iot_handlers/drivers/PrinterDriver_L.py:186]: note: by default cups already prints in the recto/verso aka "duplex" mode so this doesn't modify the behaviour

**Review by Ysoroko — APPROVED** [review]

**Comment by Ysoroko** [comment]
@robodoo r+

**Pull request closed without merging**